Source/Core/Core/IOS/USB/Host.cpp

```diff
@@ -115,6 +116,19 @@ bool USBHost::UpdateDevices(const bool always_add_hooks)
 bool USBHost::AddNewDevices(std::set<u64>& new_devices, DeviceChangeHooks& hooks,
                             const bool always_add_hooks)
 {
+  if (Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL))
+  {
+    auto skylanderportal = std::make_unique<USB::SkylanderUSB>(m_ios, "Skylander Portal");
+    if (ShouldAddDevice(*skylanderportal))
+    {
+      const u64 skyid = skylanderportal->GetId();
+      new_devices.insert(skyid);
+      if (AddDevice(std::move(skylanderportal)))
+      {
+        hooks.emplace(GetDeviceById(skyid), ChangeEvent::Inserted);
+      }
+    }
+  }
 #ifdef __LIBUSB__
```

Comment:
You'll need to make several changes to make emulated USB devices work in deterministic mode:

* in USBHost::Open and USBHost::UpdateDevices, remove the Core::WantsDeterminism() checks
* in USBHost::ScanThread::Start:
  * if Core::WantsDeterminism(), then schedule a period CoreTiming event to call m_host->UpdateDevices()
  * otherwise, spawn the USB scan thread (basically, the existing thread code needs to be moved into the else branch)
* USBHost::ScanThread::WaitForFirstScan should check if the thread is running; if it isn't, then there is no need to wait for m_first_scan_complete_event
* in USBHost::AddNewDevices, wrap the libusb code inside a Core::WantsDeterminism() check so that we don't expose real USB devices in e.g. Movie mode

Reply:
Instead of removing the existing Core::WantsDeterminism checks, what would the consequence be for just creating an else clause in the USBHost::Open method to call AddEmulatedDevices once?